File: app/src/main/java/jp/gr/java_conf/snake0394/loglook_android/bean/MstSlotitem.java
package jp.gr.java_conf.snake0394.loglook_android.bean;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

/**
 * 装備品のデータを持つクラス
 */
public class MstSlotitem implements Serializable {

    private static final long serialVersionUID = -4407183149378737644L;

    /**
     * id
     */
    @SerializedName("api_id")
    private Integer id;

    /**
     * 名前
     */
    @SerializedName("api_name")
    private String name;

    /**
     * 並び替え順
     */
    @SerializedName("api_sortno")
    private int sortno;

    /**
     * 装備タイプ [0]=大分類、[1]=図鑑表示、[2]=カテゴリ、[3]=アイコンID
     */
    @SerializedName("api_type")
    private List<Integer> type;

    /**
     * 耐久(0)
     */
    @SerializedName("api_taik")
    private int taik;

    /**
     * 装甲
     */
    @SerializedName("api_souk")
    private int souk;

    /**
     * 火力
     */
    @SerializedName("api_houg")
    private int houg;

    /**
     * 雷装
     */
    @SerializedName("api_raig")
    private int raig;

    /**
     * 速力
     */
    @SerializedName("api_soku")
    private int soku;

    /**
     * 爆装
     */
    @SerializedName("api_baku")
    private int baku;

    /**
     * 対空
     */
    @SerializedName("api_tyku")
    private int tyku;

    /**
     * 対潜
     */
    @SerializedName("api_tais")
    private int tais;

    /**
     * ?(0)
     */
    @SerializedName("api_atap")
    private int atap;

    /**
     * 命中
     */
    @SerializedName("api_houm")
    private int houm;

    /**
     * 雷撃命中
     */
    @SerializedName("api_raim")
    private int raim;

    /**
     * 回避
     */
    @SerializedName("api_houk")
    private int houk;

    /**
     * 雷撃回避(0)
     */
    @SerializedName("api_raik")
    private int raik;

    /**
     * 爆撃回避(0)
     */
    @SerializedName("api_bakk")
    private int bakk;

    /**
     * 索敵
     */
    @SerializedName("api_saku")
    private int saku;

    /**
     * 索敵妨害(0)
     */
    @SerializedName("api_sakb")
    private int sakb;

    /**
     * 運(0)
     */
    @SerializedName("api_luck")
    private int luck;

    /**
     * 射程
     */
    @SerializedName("api_leng")
    private int leng;

    /**
     * レアリティ
     */
    @SerializedName("api_rare")
    private int rare;

    /**
     * 廃棄資材
     */
    @SerializedName("api_broken")
    private List<Integer> broken;

    /**
     * 図鑑情報
     */
    @SerializedName("api_info")
    private String info;

    /**
     * ?
     */
    @SerializedName("api_usebull")
    private String usebull;

    /**
     * 航空機コスト
     */
    @SerializedName("api_cost")
    private int cost;

    /**
     * 航続距離
     */
    @SerializedName("api_distance")
    private int distance;


    public MstSlotitem() {
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getSortno() {
        return sortno;
    }

    public void setSortno(int sortno) {
        this.sortno = sortno;
    }

    public List<Integer> getType() {
        return type;
    }

    public void setType(List<Integer> type) {
        this.type = type;
    }

    public int getTaik() {
        return taik;
    }

    public void setTaik(int taik) {
        this.taik = taik;
    }

    public int getSouk() {
        return souk;
    }

    public void setSouk(int souk) {
        this.souk = souk;
    }

    public int getHoug() {
        return houg;
    }

    public void setHoug(int houg) {
        this.houg = houg;
    }

    public int getRaig() {
        return raig;
    }

    public void setRaig(int raig) {
        this.raig = raig;
    }

    public int getSoku() {
        return soku;
    }

    public void setSoku(int soku) {
        this.soku = soku;
    }

    public int getBaku() {
        return baku;
    }

    public void setBaku(int baku) {
        this.baku = baku;
    }

    public int getTyku() {
        return tyku;
    }

    public void setTyku(int tyku) {
        this.tyku = tyku;
    }

    public int getTais() {
        return tais;
    }

    public void setTais(int tais) {
        this.tais = tais;
    }

    public int getAtap() {
        return atap;
    }

    public void setAtap(int atap) {
        this.atap = atap;
    }

    public int getHoum() {
        return houm;
    }

    public void setHoum(int houm) {
        this.houm = houm;
    }

    public int getRaim() {
        return raim;
    }

    public void setRaim(int raim) {
        this.raim = raim;
    }

    public int getHouk() {
        return houk;
    }

    public void setHouk(int houk) {
        this.houk = houk;
    }

    public int getRaik() {
        return raik;
    }

    public void setRaik(int raik) {
        this.raik = raik;
    }

    public int getBakk() {
        return bakk;
    }

    public void setBakk(int bakk) {
        this.bakk = bakk;
    }

    public int getSaku() {
        return saku;
    }

    public void setSaku(int saku) {
        this.saku = saku;
    }

    public int getSakb() {
        return sakb;
    }

    public void setSakb(int sakb) {
        this.sakb = sakb;
    }

    public int getLuck() {
        return luck;
    }

    public void setLuck(int luck) {
        this.luck = luck;
    }

    public int getLeng() {
        return leng;
    }

    public void setLeng(int leng) {
        this.leng = leng;
    }

    public int getRare() {
        return rare;
    }

    public void setRare(int rare) {
        this.rare = rare;
    }

    public List<Integer> getBroken() {
        return broken;
    }

    public void setBroken(List<Integer> broken) {
        this.broken = broken;
    }

    public String getInfo() {
        return info;
    }

    public void setInfo(String info) {
        this.info = info;
    }

    public String getUsebull() {
        return usebull;
    }

    public void setUsebull(String usebull) {
        this.usebull = usebull;
    }

    public int getCost() {
        return cost;
    }

    public void setCost(int cost) {
        this.cost = cost;
    }

    public int getDistance() {
        return distance;
    }

    public void setDistance(int distance) {
        this.distance = distance;
    }

}

File: app/src/main/java/jp/gr/java_conf/snake0394/loglook_android/bean/MstSlotitem.kt
package jp.gr.java_conf.snake0394.loglook_android.bean

import com.google.gson.annotations.SerializedName
import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass
import jp.gr.java_conf.snake0394.loglook_android.storage.RealmInt

/**
 * 装備品のデータを持つクラス
 */
@RealmClass
open class MstSlotitem(
        /**
         * id
         */
        @PrimaryKey
        @SerializedName("api_id")
        open var id: Int = -1,

        /**
         * 名前
         */
        @SerializedName("api_name")
        open var name: String = "",

        /**
         * 並び替え順
         */
        @SerializedName("api_sortno")
        open var sortno: Int = -1,

        /**
         * 装備タイプ [0]=大分類、[1]=図鑑表示、[2]=カテゴリ、[3]=アイコンID
         */
        @SerializedName("api_type")
        open var type: RealmList<RealmInt> = RealmList(),

        /**
         * 耐久(0)
         */
        @SerializedName("api_taik")
        open var taik: Int = 0,

        /**
         * 装甲
         */
        @SerializedName("api_souk")
        open var souk: Int = 0,

        /**
         * 火力
         */
        @SerializedName("api_houg")
        open var houg: Int = 0,

        /**
         * 雷装
         */
        @SerializedName("api_raig")
        open var raig: Int = 0,

        /**
         * 速力
         */
        @SerializedName("api_soku")
        open var soku: Int = 0,

        /**
         * 爆装
         */
        @SerializedName("api_baku")
        open var baku: Int = 0,

        /**
         * 対空
         */
        @SerializedName("api_tyku")
        open var tyku: Int = 0,

        /**
         * 対潜
         */
        @SerializedName("api_tais")
        open var tais: Int = 0,

        /**
         * ?(0)
         */
        @SerializedName("api_atap")
        open var atap: Int = 0,

        /**
         * 命中
         */
        @SerializedName("api_houm")
        open var houm: Int = 0,

        /**
         * 雷撃命中
         */
        @SerializedName("api_raim")
        open var raim: Int = 0,

        /**
         * 回避
         */
        @SerializedName("api_houk")
        open var houk: Int = 0,

        /**
         * 雷撃回避(0)
         */
        @SerializedName("api_raik")
        open var raik: Int = 0,

        /**
         * 爆撃回避(0)
         */
        @SerializedName("api_bakk")
        open var bakk: Int = 0,

        /**
         * 索敵
         */
        @SerializedName("api_saku")
        open var saku: Int = 0,

        /**
         * 索敵妨害(0)
         */
        @SerializedName("api_sakb")
        open var sakb: Int = 0,

        /**
         * 運(0)
         */
        @SerializedName("api_luck")
        open var luck: Int = 0,

        /**
         * 射程
         */
        @SerializedName("api_leng")
        open var leng: Int = 0,

        /**
         * レアリティ
         */
        @SerializedName("api_rare")
        open var rare: Int = 0,

        /**
         * 廃棄資材
         */
        @SerializedName("api_broken")
        open var broken: RealmList<RealmInt> = RealmList(),

        /**
         * 図鑑情報
         */
        @SerializedName("api_info")
        open var info: String = "",

        /**
         * ?
         */
        @SerializedName("api_usebull")
        open var usebull: String = "",

        /**
         * 航空機コスト
         */
        @SerializedName("api_cost")
        open var cost: Int = 0,

        /**
         * 航続距離
         */
        @SerializedName("api_distance")
        open var distance: Int = 0
) : RealmObject()